gcc/testsuite/rust.test/xfail_compile/unused.rs

```diff
@@ -0,0 +1,16 @@
+
+fn bar() { // {dg-warning "function is never used: `bar`"}
```

Comment:
@dkm do you think this test case should be moved into compile? Since this is simply about warnings

Reply:
I guess so yes. `xfail` should really be about compiler returning with an error

Reply:
I had moved unused.rs to compile dir.

Reply:
> Okay, is there any chance to create a new test dir for specific warning outputs? Because I want to make sure that the test will fail if the dg-warning doesn't match the real compiler warning.

We don't need a new category/directory for that; it should already behave in that way (have you checked?) -- if not, that's a bug in the test harness.